File: solutions/typescript/word-count/1/word-count.ts
export function count(subtitle: string) {
  const counts = new Map<string, number>();
  const words = subtitle.toLowerCase().match(/[a-z0-9]+('[a-z0-9]+)?/g);
  
  if (words) {
    for (const word of words) {
      counts.set(word, (counts.get(word) || 0) + 1);
    }
  }
  
  return counts;
}

File: solutions/julia/word-count/1/word-count.jl
function wordcount(sentence)

    lowercase_sentence = lowercase(sentence)
    

    words = map(m -> m.match, eachmatch(r"\w+('\w+)?", lowercase_sentence))
    
 
    counts = Dict{String, Int}()
    for word in words
        counts[word] = get(counts, word, 0) + 1
    end
    
    return counts
end
